notebooks/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,660 @@
+   },
+   "outputs": [],
+   "source": [
+    "train_data[\"family\"].describe()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>The only categorical feature (in this table) \"family\" has 33 possible unique values (not so a lot) <br>\n",
+    "That means we can easily use one-hot-encoding during model training</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "family_values = train_data[\"family\"].unique()\n",
+    "family_values"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 40))\n",
+    "for i, value in zip(range(len(family_values)), family_values):\n",
+    "  plt.subplot(len(family_values)//3, 3, i+1)\n",
+    "  current_value_data = train_data[train_data[\"family\"] == value].groupby([\"date\"]).mean()\n",
+    "  plt.xlim=(0, 1750)\n",
+    "  plt.ylim=(0, current_value_data[\"sales\"].max())\n",
+    "  plt.scatter(x=np.arange(len(current_value_data.index)), y=current_value_data[\"sales\"])\n",
+    "  plt.title(value)\n",
+    "plt.xlim=(0, 1750)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>Actually, these plots don't really help us, because, looking on them, we can only catch some common \"trends\", which describe global changes in some products sales.<br>For example, we can be sure that on the 1th January every year number of sales is equal to 0 (actually, it is necessary to check, but I think it is obvious). <br> Also we can say that the number of sales in general has a positive dynamic (for example, 'automotive', 'bread/bakery', 'grocery i', 'personal care', ...).<br>Some of the product types have a negative dynamic (such as lingerie).<br>Some goods have very interesting sales distribution (books, produce, froxen foods, ladieswear, ...), and we can't say right now, what is the reason of that.<br>Also some goods families have seasonal increase in sales('school and office supplies', 'liquor, wine, beer', 'grocery ii', 'frozen foods'). </i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 110))\n",
+    "for i, family_value in zip(range(len(family_values)), family_values):\n",
+    "    plt.subplot(17, 2, i+1)\n",
+    "    average_sales = train_data[train_data['family'] == family_value].groupby('date').mean()['sales']\n",
+    "    trend = average_sales.rolling(\n",
+    "        window=365,\n",
+    "        center=True,\n",
+    "        min_periods=183,\n",
+    "    ).mean()\n",
+    "    ax = average_sales.plot(alpha=0.5)\n",
+    "    ax = trend.plot(ax=ax, linewidth=3)\n",
+    "    plt.title(family_value)\n",
```

Comment:
Let's be a bit more explicit with matplotlib axes (previous comments are also applicable here):
```suggestion
    "    ax = plt.subplot(17, 2, i + 1)\n",
    "    average_sales = train_data[train_data['family'] == family_value].groupby('date')['sales'].mean()\n",
    "    trend = average_sales.rolling(\n",
    "        window=365,\n",
    "        center=True,\n",
    "        min_periods=183,\n",
    "    ).mean()\n",
    "    average_sales.plot(alpha=0.5, ax=ax)\n",
    "    trend.plot(ax=ax, linewidth=3)\n",
    "    ax.set_title(family_value)\n",
```